ldes-server-infra-mongo/mongo-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/infra/mongo/fragment/LdesFragmentMongoRepository.java

```diff
@@ -99,12 +99,16 @@ public void incrementNumberOfMembers(String fragmentId) {
 	}
 
 	@Override
-	public List<LdesFragment> retrieveFragmentsOfView(String viewName) {
+	public Stream<LdesFragment> retrieveFragmentsOfView(String viewName) {
```

Comment:
I feel like we lose the power of Stream because we first do a 'findAll' to a list and then stream it. If I remember correctly, MongoRepository supports 'streamAllBy' as well. Then it manages a cursor for you and you are actually streaming the data from the database.
Note: this comment is for all the Stream changes in this PR

Reply:
Top comment!